index.d.ts

```diff
@@ -0,0 +1,10 @@
+interface ClearablePromise<T> extends Promise<T> {
+  clear (): void
+}
+
+declare const delay: {
+  <T = never>(ms: number, value?: T): ClearablePromise<T>
+  reject(ms: number, reason?: any): ClearablePromise<never>
```

Comment:
Why is this not also `T`?

Reply:
I found the relevant TS issue: https://github.com/Microsoft/TypeScript/issues/5413 Can you add a TODO comment about making it more strict when that issue is fixed?